Implementation/FindMyBLEDevice/FindMyBLEDevice/ViewModels/ItemDetailViewModel.cs

```diff
@@ -59,15 +61,32 @@ public void UserLabel_TextChanged()
 
         async Task RenameDevice()
         {
+            
+            // Show confirmation dialog
+            bool answer = await Application.Current.MainPage.DisplayAlert("Rename device", String.Format("Are you sure you want to rename this device to '{0}'?", UserLabel), "Yes", "Cancel");
+
+            if (!answer)
+            {
```

Comment:
you might want to reset the textbox to the old name here

Reply:
Thanks for the advice, I implemented it! :)